com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -573,29 +578,41 @@ internal void PostNetworkVariableWrite()
             }
         }
 
-        internal void VariableUpdate(ulong clientId)
+        internal void VariableUpdate(ulong clientId, SnapshotSystem snapshot)
         {
             if (!m_VarInit)
             {
                 InitializeVariables();
             }
 
             PreNetworkVariableWrite();
-            NetworkVariableUpdate(clientId);
+            NetworkVariableUpdate(clientId, NetworkBehaviourId, snapshot);
         }
 
         private readonly List<int> m_NetworkVariableIndexesToReset = new List<int>();
         private readonly HashSet<int> m_NetworkVariableIndexesToResetSet = new HashSet<int>();
 
-        private void NetworkVariableUpdate(ulong clientId)
+        private void NetworkVariableUpdate(ulong clientId, int behaviourIndex, SnapshotSystem snapshot)
         {
             if (!CouldHaveDirtyNetworkVariables())
             {
                 return;
             }
 
+            if (s_UseSnapshot)
+            {
+                for (int k = 0; k < NetworkVariableFields.Count; k++)
+                {
+                    snapshot.Store(NetworkObjectId, behaviourIndex, k, NetworkVariableFields[k]);
+                }
+            }
+
             for (int j = 0; j < m_ChannelMappedNetworkVariableIndexes.Count; j++)
             {
+                if (!s_UseClassicDelta)
```

Comment:
Can this 'if' be hoisted outside the for loop?  s_UseClassicDelta is loop-invariant

Reply:
Only downside is that we now touch 100 lines of code by indenting them. But yes, long-term readability trumps merge size.